### PR TITLE
Update phpunit.xml based on generate config by PHPUnit

### PR DESCRIPTION
**Pull request by yguedidi**
extracted from #3576, without the .dist part, in case it passes alone

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
-      <directory>utils-tests</directory>
-    </testsuite>
-  </testsuites>
-  <php>
-    <ini name="memory_limit" value="-1"/>
-    <env name="XDEBUG_MODE" value="coverage"/>
-  </php>
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.0/phpunit.xsd"
```
Comment (yguedidi) [inline phpunit.xml:3]: added valid schema

File: phpunit.xml [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+         bootstrap="tests/bootstrap.php"
+         cacheDirectory=".phpunit.cache"
+         executionOrder="depends,defects"
```
Comment (yguedidi) [inline phpunit.xml:7]: depends before defects

File: phpunit.xml [hunk 3]
```diff
@@ -1,15 +1,23 @@
-<?xml version="1.0"?>
-<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd" bootstrap="tests/bootstrap.php" colors="true" executionOrder="defects" cacheDirectory=".phpunit.cache">
```
Comment (yguedidi) [inline phpunit.xml:2]: ~colors=true is the default~ wrong: https://docs.phpunit.de/en/10.0/configuration.html#the-colors-attribute

File: phpunit.xml [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
-  </php>
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.0/phpunit.xsd"
+         bootstrap="tests/bootstrap.php"
+         cacheDirectory=".phpunit.cache"
+         executionOrder="depends,defects"
+         beStrictAboutOutputDuringTests="true"
+         failOnRisky="true"
+         failOnWarning="true">
```
Comment (yguedidi) [inline phpunit.xml:10]: kept those from recommended config, removed:

```
requireCoverageMetadata="true"
beStrictAboutCoverageMetadata="true"
```

**Comment by TomasVotruba** [comment]
Could you keep the spacing? The diff is now very hard to read as completelly restarted.

**Comment by yguedidi** [comment]
> Could you keep the spacing?

@TomasVotruba actually I kept the spacing and wrapping of the recommended config generated with `--generate-configuration`.
That's why I highlighted the changes with comments.
Finals decision is yours, but I'd suggest to stick to the recommended standards from PHPUnit

**Comment by TomasVotruba** [comment]
I don't see any advantage or bug fix here, so we'll stick with current version. Thanks for understanding

**Comment by yguedidi** [comment]
Ok

**Pull request closed without merging**